### ISSUE TITLE
A few remarks

### ISSUE THREAD

**Issue by laurmaedje** [johanvx/typst-undergradmath#2]
Hey, I'm one of the Typst developers. This is great!

A few remarks:
- The thing with the script letters is interesting. Are they even part of Unicode?
- Widehat is available, it's automatic if you write `$hat(x+y)$`
- doteq is `$dot(eq)$` because the dot is an accent
- Maybe there can be an explanation for scaling: They scale by default if the symbol is entered directly as a codepoint, they don't scale automatically if you use symbol notation like `$angle.l$`
- The examples for the `h` function are missing a hashtag because the function consists only of a single letter
- The alignment example is a bit weird since it would look the same without alignment, but I guess that goes for the original LaTeX version, too
- `$"9.8" "m/s"^2$` would work for the units thing, but it's obviously hacky and not good. Typst needs to support this better.

PS: Would you be fine with us posting this on Twitter with the Typst account?

**Comment by johanvx** [comment]
Thanks for your attention! **Feel free to share and use this anywhere!**

Reply on the remarks:

> The thing with the script letters is interesting. Are they even part of Unicode?

I didn't think about this. So I've just search on Compart (https://www.compart.com/en/unicode/) and found that they **are** Unicode letters. For example, the letter $\mathscr{P}$ `\mathscr{P}` is actually [Mathematical Script Capital P](https://www.compart.com/en/unicode/U+1D4AB) `\u{1D4AB}`.

> Widehat is available, it's automatic if you write `$hat(x+y)$`

Oh, nice! I didn't even try this. I'll update the document then.

> doteq is `$dot(eq)$` because the dot is an accent

I've tried this but didn't adopt it. I don't remember why though. Maybe the spacing is not identical with the LaTeX `\doteq`? I'll check this soon.

> Maybe there can be an explanation for scaling: They scale by default if the symbol is entered directly as a codepoint, they don't scale automatically if you use symbol notation like $angle.l$.

Yeah, I think there should be some explanations. I'll add this comment by creating a co-authored commit.

> The examples for the `h` function are missing a hashtag because the function consists only of a single letter

Oh yes, mentioning single letter functions without a hashtag may be ambiguous. Thanks for pointing this out.

> The alignment example is a bit weird since it would look the same without alignment, but I guess that goes for the original LaTeX version, too

I just tried to mimic the original LaTeX version 😝 and didn't think much on this.

> $"9.8" "m/s"^2$ would work for the units thing, but it's obviously hacky and not good. Typst needs to support this better.

I have no idea here because I'm not familiar with the `~` spacing in LaTeX. I'll add this (with a 💦 annotation as this is hacky) in a co-authored commit.

**Comment by johanvx** [comment]
> I've tried this but didn't adopt it. I don't remember why though. Maybe the spacing is not identical with the LaTeX `\doteq`? I'll check this soon.

So here's a simple test.

LaTeX | Typst

![`doteq` comparison between LaTeX and Typst (from left to right), the spacing between `dot` and `eq` in Typst is larger](https://user-images.githubusercontent.com/61529310/227709269-1161b565-3f20-40bb-b560-63330d1c265a.png)

with LaTeX code

```
\documentclass[10pt]{article}
\usepackage{mathtools, amssymb, amsthm}
\begin{document}
\[ \doteq \]
\end{document}
```

and Typst code

```
#set text(10pt)
$ dot(eq) $
```

**Comment by user202729** [comment]
Remark on script characters: The font files are in a typical LaTeX distribution ([CTAN: /tex-archive/fonts/rsfs](https://www.ctan.org/tex-archive/fonts/rsfs/)), but they're postscript type 1 font, and I think Typst only support opentype font.

Maybe something in https://stackoverflow.com/questions/9322635/how-can-i-convert-postscript-type-1-font-to-otf-or-ttf can convert them to a Typst-compatible format.

**Comment by johanvx** [comment]
I wrote something weird but maybe usable.

```
#let eqdot = $\u{2A66}$
#let doteq2 = style(styles => {
  let ex = measure("x", styles).height
  box(baseline: -ex, align(center, stack(spacing: (0.3455 - 1 + 0.1874 * 2) * ex, $dot$, $eq$)))
})

$ doteq2 = eqdot quad a doteq2 b, a = b, a eqdot b \
  2^(doteq2 = eqdot quad a doteq2 b, a = b, a eqdot b) $
```

![Output of the above Typst code](https://user-images.githubusercontent.com/61529310/227758217-2fc0a409-e18b-4d54-bb20-4e3f19ac0e03.png)

Sad to find out that `eqdot` is a Unicode character ([Equals Sign with Dot Below]) while `doteq` is not.

[Equals Sign with Dot Below]: https://www.compart.com/en/unicode/U+2A66

**Comment by user202729** [comment]
See, the problem is that your doteq2 does not expect mathstyle. In superscript doteq still keeps some spacing around it, while = and eqdot doesn't.

**Comment by johanvx** [comment]
For `doteq`, see #9.

**Comment by kokic** [comment]
The width of `hat` is not always suitable e.g. `hat(x_i)` or `hat(x y)` . This has forced me to write `hat(space x_i space)`.

**Comment by johanvx** [comment]
> The width of `hat` is not always suitable e.g. `hat(x_i)` or `hat(x y)` . This has forced me to write `hat(space x_i space)`.

Good point! I hope that readers can do this with the help of Section **Spacing in mathematics**.

Just out of curiosity: in what context are these use cases?

For me:

- $\Large \widehat{x_i}$ (`hat(space x_i space)` in Typst / `\widehat{x_i}` in LaTeX) is _really rare_. Almost always use $\Large \hat{x}_i$ (`hat(x)_i`/`\hat{x}_i`).
- Never met $\Large \widehat{x y}$, but a similar thing without extra `space` $\Large (\widehat{\vec{u}, \vec{v}})$ (`(hat(arrow(u)\, arrow(v)))`[^*]/`(\widehat{\vec{u}, \vec{v}})`) is OK.

[^*]: actually, maybe something like `paren.l hat(#h(0.1em) arrow(u)\, arrow(v) #h(0.1em)) paren.r` is better.
